### ISSUE TITLE
Flask-WTF Question

### ISSUE THREAD

**Issue by samuelerickson977** [miguelgrinberg/flasky#553]
Please do `git checkout 4a` and look at `line 33` in `hello.py`. Can you explain to me why instantiating the NameForm object after every request does not overwrite the state of the form from the POST request? For example, I navigate to `/`, and then enter my name and click submit. When I click submit, the POST request uses the `index()` function. However, if `index()` is called, then won't the form object just the POST request lose it's data because a new `NameForm` object is instantiated?

**Comment by miguelgrinberg** [comment]
No, that's not how Flask-WTF works. When you create the `NameForm` class Flask-WTF looks in the request object from Flask to see if data for the form has been submitted in the current request. If yes, it loads the data into the newly created form object. If not it creates the form object without data. The form data is not persisted anywhere, you have to save it (or use it somehow) before the request that handles the form submission ends.

**Comment by samuelerickson977** [comment]
OK, that makes more sense. Thank you very much sir.
